types/tx/signing/signature.go

```diff
@@ -15,3 +19,64 @@ type SignatureV2 struct {
 	// the signatures themselves for either single or multi-signatures.
 	Data SignatureData
 }
+
+// SignatureDataToSignatureDescriptorData converts a SignatureData to SignatureDescriptor_Data.
+// SignatureDescriptor_Data is considered an encoding type whereas SignatureData is used for
+// business logic.
+func SignatureDataToSignatureDescriptorData(data SignatureData) *SignatureDescriptor_Data {
```

Comment:
TBH I would be happy to remove the `SignatureData` interface (which is just an abstraction to represent `oneof SingleSignatureData | MultiSignatureData`), and just use the concrete type SignatureDescriptor everywhere in the code

Reply:
I do agree here. I'd go with what @amaurymartiny stated or at the very least, change the nomenclature because it's confusing and the types clobber each other.

Reply:
What if for the time-being we rename this function to `SignatureDataToProto` or something like that?

And then maybe in a future PR we just have a single proto generated `SignatureData` that is used everywhere as @amaurymartiny is suggesting?

This is an example of different encoding/application types btw. I've always questioned that philosophy because of the amount of duplication... Anyway, I think this is a concrete example of that.

Reply:
SignatureDataToProto is bit confusing as well. I suggest to keep it as it is now and unblock this PR, and address this in a followup PR as @amaurymartiny suggesting

Reply:
Is that okay @alexanderbez ? It would be good to get a decision either way because this is one of the last critical pieces to enabling proto tx as the default.

Reply:
If we leave it as-is, it's super confusing IMHO. I'd prefer `SignatureDataToProto` TBH. You're converting a domain-type to a wire-type (i.e. Proto).

Reply:
Okay, so how about we go with `SignatureDataTo/FromProto` here? And as a follow-up @amaurymartiny you'll take a look at possible improvements?

Reply:
that sgtm 👍